fv/workload/workload.go

```diff
@@ -399,6 +413,35 @@ func (c *ConnectivityChecker) ExpectedConnectivity() []string {
 	return result
 }
 
-func (c *ConnectivityChecker) CheckConnectivity() {
-	EventuallyWithOffset(1, c.ActualConnectivity(), "10s", "100ms").Should(Equal(c.ExpectedConnectivity()))
+func (c *ConnectivityChecker) CheckConnectivity(optionalDescription ...interface{}) {
+	// Make sure that we retry at least once even if the check itself takes >10s.
+	expConnectivity := c.ExpectedConnectivity()
+	if reflect.DeepEqual(c.ActualConnectivity(), expConnectivity) {
+		return
+	}
+	start := time.Now()
+	completedAttempts := 0
+	var actualConn []string
+	for time.Since(start) < 10*time.Second || completedAttempts <= 1 {
```

Comment:
Think || should be && here.

Reply:
It's correct now I removed the extra connectivity check at the start of the function.  The intention is to make sure that we run a minimum of two checks if the first one fails.

Reply:
OK - but the problem is that two checks might completely successfully in less than 10s, and then the loop will continue iterating because 10s have not elapsed yet.  Surely that is wrong?

Reply:
It does an early return if the connectivity is ever correct.

Reply:
Ah, right, I get it now: "**If the first connectivity check fails**, we want [to do at least one retry]".